distributed/scheduler.py

```diff
@@ -1565,6 +1570,30 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    @property
+    def published_data_keys(self):
+        return list(sorted(self.published_data.keys()))
+
+    def publish_data(self, keys=None, data=None, name=None):
+        self.published_data[name] = {'data': pickle.loads(data),
+                                     'keys': pickle.loads(keys)}
+        # TODO: add guards against deletion of keys?
+
+    def get_published_keys(self, *args):
+        return self.published_data_keys
+
+    def get_published_data(self, stream, name=None, client=None):
+        keys = [str(k) for k in self.published_data[name]['keys']]
+        self.client_wants_keys(keys=keys, client=client)
+        return pickle.dumps(self.published_data[name])
+
+    def poll_keys(self, stream, keys=None):
+        keys = pickle.loads(keys)
+        for key in keys:
+            key = str(key)
+            if self.task_state[key] in ('memory', 'erred'):
+                self.report_on_key(key)
```

Comment:
The line `self.client_wants_keys(keys=keys, client=client)` does this above, during the first call. I did play with update-graph too, but didn't get anywhere (yet)